src/Worker/WorkerResolver.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Worker;
+
+use Spiral\RoadRunner\Environment\Mode;
+use Spiral\RoadRunner\EnvironmentInterface;
+
+final class WorkerResolver implements WorkerResolverInterface
+{
+    private EnvironmentInterface $environment;
+    private HttpWorker $httpWorker;
+    private TemporalWorker $temporalWorker;
+
+    public function __construct(
+        EnvironmentInterface $environment,
+        HttpWorker $httpWorker,
+        TemporalWorker $temporalWorker
```

Comment:
Doing that forces the DIC to resolve the 2 workers but only one is used (and Temporal might even be not installed).

It's the kind of place where we could inject the container and use `return $container->get(HttpWorkerInferface::class)` so the service is created only if needed.

Reply:
I've marked them with `lazy()`

Reply:
I don't like service locator. As you can see :)

What do you think if I add `supports($mode): bool` method into workers and configure resolver to get array of workers and pass the conditions below into them?
It will look like the following:
```
TemporalWorker:
supports($mode):
$mode === 'temporal'


HttpWorker:
supports($mode):
$mode === 'http'
```

And this resolver will look:
```php
class WorkerResolver
{
  function __construct(iterable $workers){}

  function resolve($mode) { 
    foreach($this->workers as $worker)) {
       if($worker->supports($mode){
         return $worker;
       }
    throw new Exception();
}
```

WorkerResolver will be configured in the container with a `tag` and `!tagged_iterator`.

Reply:
If you do this, the DIC may still instantiate the 2 workers but only one is useful, I think a locator is the best way, it's precisily why service locators exists, in the Symfony doc: 

> Sometimes, a service needs access to several other services without being sure that all of them will actually be used